MM-11477 Changes required to intercept leaked network errors on mobile

hmhealey opened the pull request:
Somewhere, we're not properly catching the error objects thrown by Client4 which has been causing them to be logged by Sentry for months, but due to how Sentry handles things that are thrown that aren't actual errors, it provides no information about where we're not handling those.

By using a modified version of the thunk middleware on mobile, we'll be able to intercept those events and replace them with more useful logs on Sentry.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-11477

jwilander commented on src/store/middleware.js:
```diff
@@ -0,0 +1,48 @@
+additionalMiddleware - func | array - Allows for single or multiple additional middleware functions to be passed in from the client side.
+enableBuffer - bool - default = true - If true, the store will buffer all actions until offline state rehydration occurs.
+enableThunk - bool - default = true - If true, include the thunk middleware automatically. If false, thunk must be provided as part of additionalMiddleware.
+***/
```
Does this comment make sense here?

jwilander commented on src/store/configureStore.dev.js:
```diff
@@ -28,40 +25,21 @@ const devToolsEnhancer = (
-It has two properties currently:
-enableBuffer - bool - default = true - If true the store will buffer all actions until offline state rehydration occurs.
-additionalMiddleware - func | array - Allows for single or multiple additional middleware functions to be passed in from the client side.
-***/
```
Why remove the comments?

hmhealey commented:
@jwilander I moved those comments back to the configureStore methods and I added the other parameters to them so that they document more than just the settings used to configure middleware
